src/main/java/uk/gov/pay/publicauth/app/PublicAuthApp.java

```diff
@@ -82,19 +83,27 @@ public void run(PublicAuthConfiguration conf, Environment environment) {
         environment.jersey().register(new HealthCheckResource(environment));
         environment.jersey().register(new ValidationExceptionMapper());
         environment.jersey().register(new TokenNotFoundExceptionMapper());
-        environment.healthChecks().register("database", new DatabaseHealthCheck(conf.getDataSourceFactory(), environment, "publicauth"));
+        environment.healthChecks().register("database", new DatabaseHealthCheck(conf.getDataSourceFactory()));
 
         environment.servlets().addFilter("LoggingFilter", new LoggingFilter())
                 .addMappingForUrlPatterns(of(REQUEST), true, "/v1" + "/*");
     }
 
     private void initialiseMetrics(PublicAuthConfiguration configuration, Environment environment) {
+        DatabaseMetricsService metricsService = new DatabaseMetricsService(configuration.getDataSourceFactory(), environment.metrics(), "publicauth");
+
+        environment
+                .lifecycle()
+                .scheduledExecutorService("metricscollector")
+                .threads(1)
+                .build()
+                .scheduleAtFixedRate(metricsService::updateMetricData, 0, GRAPHITE_SENDING_PERIOD_SECONDS / 2, TimeUnit.SECONDS);
```

Comment:
I figured we'd want to ensure that every time we send to graphite we've definitely got fresh stats. It could be `GRAPHITE_SENDING_PERIOD_SECONDS - 1` I suppose. It was a figure plucked from the air somewhat